core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java

```diff
@@ -0,0 +1,84 @@
+package ca.jrvs.apps.grep;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Java regex scanner. Recursively scans the given directory, looking for filenames which match
+ * the supplied regex. The results of the scan are written to the supplied output file.
+ */
+public class JavaGrepLambdaImp extends JavaGrepImp {
+
+  private Logger jgrepLogger;
+
+  /**
+   * JavaGrepImp class constructor.
+   */
+  public JavaGrepLambdaImp() {
+    jgrepLogger = LoggerFactory.getLogger(JavaGrepLambdaImp.class);
+  }
+
+  /**
+   * Overrides JavaGrepImp.readLines(). This implementation uses java.nio.Files.
+   * @param file a File object which lines must be read from. This must be a file, not a directory
+   * @return a List of Strings containing all readable lines of the supplied file
+   */
+  @Override
+  public List<String> readLines(File file) {
+    List<String> lines = new ArrayList<>();
+    try {
+      lines = Files.readAllLines(file.toPath()); // Ed may not consider this "using streams"
+    } catch (IOException ex) {
```

Comment:
You are right, this is not.using streams.
Try again by creating a stream of just the one file.  Look up how to use tbe flatmap function to turn the file into a stream of strings (lines in the file).

Reply:
changed to `Files.lines(file.toPath()).collect(Collectors.toList());`